pdr_backend/util/subgraph_predictions.py

```diff
@@ -0,0 +1,211 @@
+                            token {{
+                                id
+                                name
+                                nft{{
+                                    nftData {{
+                                    key
+                                    value
+                                    }}
+                                }}
+                            }}
+                        }}
+                    }}
+                }}
+            }}"""
+
+        # print("Querying subgraph...", query)
+        result = query_subgraph(
+            get_subgraph_url(network),
+            query,
+            timeout=20.0,
+        )
+
+        offset += chunk_size
+
+        if not "data" in result:
+            break
+
+        data = result["data"]["predictPredictions"]
+        if len(data) == 0:
+            break
+
+        for prediction in data:
+            info725 = prediction["slot"]["predictContract"]["token"]["nft"]["nftData"]
+            info = info_from_725(info725)
+            pair_name = info["pair"]
+            timeframe = info["timeframe"]
+            source = info["source"]
+            timestamp = prediction["slot"]["slot"]
+
+            if prediction["payout"] is None:
+                continue
+
+            trueval = prediction["payout"]["trueValue"]
+            payout = float(prediction["payout"]["payout"])
+
+            if trueval is None:
+                continue
+
+            predictedValue = prediction["payout"]["predictedValue"]
+            stake = float(prediction["stake"])
+            predictoor_user = prediction["user"]["id"]
+
+            if stake < 0.01:
+                continue
+
+            prediction_obj = Prediction(
+                pair_name,
+                timeframe,
+                predictedValue,
+                stake,
+                trueval,
+                timestamp,
+                source,
+                payout,
+                predictoor_user,
+            )
+            predictions.append(prediction_obj)
+
+    return predictions
+
+
+def get_all_contracts(owner_address: str, network: str) -> List[str]:
+    if network != "mainnet" and network != "testnet":
+        raise Exception("Invalid network, pick mainnet or testnet")
+
+    # Define the GraphQL query
+    query = (
+        """
+        {
+            tokens(where: {
+                nft_: {
+                    owner: "%s"
+                }
+            }) {
+                id
+            }
+        }
+        """
+        % owner_address
+    )
+
+    # Define the subgraph endpoint
+    result = query_subgraph(get_subgraph_url(network), query, timeout=20.0)
+
+    if not "data" in result:
+        raise Exception("Error fetching contracts: No data returned")
+
+    # Parse the results and construct Contract objects
+    contract_data = result["data"]["tokens"]
+    contracts = [contract["id"] for contract in contract_data]
+
+    return contracts
+
+
+def get_contract_informations(
```

Comment:
Need a better label. "Get information" is vague.

Reply:
renamed as "fetch_contract_id_and_spe"